truffleruby/src/main/ruby/core/truffle/cext.rb

```diff
@@ -958,7 +958,22 @@ def rb_funcallv_public(recv, meth, args)
   end
 
   def rb_funcallv(recv, meth, args)
-    recv.send(meth, *args)
+    rb_funcall(recv, meth, *args)
+  end
+
+  def rb_funcall(recv, meth, *args)
+    old_c_block = Thread.current[:__C_BLOCK__]
+    begin
+      block = Thread.current[:__C_BLOCK__]
```

Comment:
The 2 variables are essentially just aliases of each other, so only one is needed.